### ISSUE TITLE
failed() for first-level failures

### ISSUE THREAD

**Issue by krlmlr** [ropensci/drake#358]
How can I quickly rerun all failed targets which haven't failed because of some upstream dependency? I want `--keep-going` (because otherwise `make()` seems to be stopping too early) but I don't want to run the entire plan.

I'd need `failed()` minus those targets where at least one dependency is also `failed()`. Maybe it's worthwhile to add an optional argument to `failed()`?

**Comment by wlandau** [comment]
Maybe this:

```r
config <- drake_config(your_plan)
next_targets <- intersect(next_stage(config), failed(cache = config$cache))
make(your_plan, targets = next_targets)
```

I think it should work because all failed targets should be invalidated.

**Comment by wlandau** [comment]
Once we transition to [`workers`](https://github.com/wlandau/workers) and we no longer need staged parallelism, there may not be a reason to have `next_stage()`. It seems straightforward to add and an `upstream_only` argument to `failed()`.

**Comment by wlandau** [comment]
Easy fix: #359

**Comment by krlmlr** [comment]
Thanks a lot! Works as advertised, saves **a lot** of hassle!
